src/v/raft/consensus.cc

```diff
@@ -1646,9 +1649,13 @@ consensus::disk_append(model::record_batch_reader&& reader) {
              std::move(reader),
              _log.make_appender(cfg),
              cfg.timeout)
-      .then([this](std::tuple<ret_t, std::vector<offset_configuration>> t) {
-          _disk_append.broadcast();
+      .then([this, should_update_last_quorum_idx](
+              std::tuple<ret_t, std::vector<offset_configuration>> t) {
           auto& [ret, configurations] = t;
```

Comment:
Add a comment that we have to update `_last_quorum_replicated_index` before triggering recovery

Reply:
salient condition 1.